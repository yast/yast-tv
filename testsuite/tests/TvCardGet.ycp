{

    include "testsuite.ycp";

    map READ = $[
	"etc" : $[ "install_inf" : $[] ],
	"target" : $[ "ycp" : $[] ],
	"probe" : $[ "system" : []],
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

    define TEST_It_Now(integer card_no_totest) ``{
	// Tv::cards == nil
	Tv::cards = nil;
	DUMP(Tv::CardGet(card_no_totest));

	// Tv::cards == []
	Tv::cards = [];
	DUMP(Tv::CardGet(card_no_totest));

	// Tv::cards == card1
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ]
	    ];
	DUMP(Tv::CardGet(card_no_totest));

	// Tv::cards == card1, card2
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ],
	      $[ "module" : "bttv",
		 "parameters" : $[ "par_other" : "testvalue2" ],
		 "unique_key" : "j7F3.J8dhgs34fK2",
		 "sound_card_no" : 1
	      ]
	    ];
	DUMP(Tv::CardGet(card_no_totest));

	// Tv::cards == card1, nil, nil, card4
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ],
	      nil,
	      nil,
	      $[ "module" : "xyz",
		 "parameters" : $[ "par_other" : "testvalue2" ],
		 "unique_key" : "j7F3.J8dhgs34fK2",
		 "sound_card_no" : 1
	      ]
	    ];
	DUMP(Tv::CardGet(card_no_totest));
    }

    TEST_It_Now(nil);

    TEST_It_Now(-1);

    TEST_It_Now(0);

    TEST_It_Now(1);

    TEST_It_Now(3);
}
