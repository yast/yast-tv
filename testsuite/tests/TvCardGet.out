Read	.target.size "/etc/install.inf" nil
Read	.etc.install_inf.Sourcemounted nil
Read	.etc.install_inf.AutoYaST nil
Read	.etc.install_inf.Demo nil
Read	.etc.install_inf.Textmode nil
Read	.etc.install_inf.Manual nil
Read	.etc.install_inf.Console nil
Read	.etc.install_inf.Braille nil
Read	.etc.install_inf.VNC nil
Read	.etc.install_inf.UseSSH nil
Read	.etc.install_inf.Locale nil
Read	.etc.install_inf.InstMode nil
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Dump	nil
Dump	nil
Dump	nil
Dump	nil
Dump	nil
Dump	nil
Dump	nil
Dump	nil
Dump	nil
Dump	nil
Dump	nil
Dump	nil
Dump	$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"]
Dump	$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"]
Dump	$["module":"bttv", "parameters":$["param1":"testvalue"], "sound_card_no":0, "unique_key":"hluk.HvsIesmXXX1"]
Dump	nil
Dump	nil
Dump	nil
Dump	$["module":"bttv", "parameters":$["par_other":"testvalue2"], "sound_card_no":1, "unique_key":"j7F3.J8dhgs34fK2"]
Dump	nil
Dump	nil
Dump	nil
Dump	nil
Dump	nil
Dump	$["module":"xyz", "parameters":$["par_other":"testvalue2"], "sound_card_no":1, "unique_key":"j7F3.J8dhgs34fK2"]
